Week9Serialization/Week9Serialization/DataContractSerialize.cs

```diff
@@ -0,0 +1,88 @@
+﻿using KellermanSoftware.CompareNetObjects;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Xml;
+using System.Runtime.Serialization;
+using System.Collections;
+
+namespace Week9Serialization
+{
+    /// <summary>
+    /// Uses DataContracts to serialize, deserialize, and compare data.
+    /// </summary>
+    public class DataContractSerialize : ISerializer
+    {
+        /// <summary>
+        /// Uses ISerializer interface to serialize data using DataContracts.
+        /// </summary>
+        /// <param name="serialList"></param>
+        static void SerializeList(VehicleList<Vehicle> serialList)
+        {
+            DataContractSerializer serialize = new DataContractSerializer(typeof(VehicleList<Vehicle>));
+
+            XmlWriterSettings settings = new XmlWriterSettings() { Indent = true };
+            using (XmlWriter writer = XmlWriter.Create("_dc-vehicles.xml", settings))
+            {
+                serialize.WriteObject(writer, serialList);
+            }
+            Console.WriteLine("Your files has been serialized.");
+        }
+        /// <summary>
+        /// Uses ISerializer interface to deserialize data using DataContracts.
+        /// </summary>
+        /// <param name="serialList"></param>
+        /// <returns></returns>
+        static VehicleList<Vehicle> DeserializeList()
+        {
+            DataContractSerializer serializer = new DataContractSerializer(typeof(VehicleList<Vehicle>));
+            VehicleList<Vehicle> deserializedList = null;
+
+            using (XmlReader reader = XmlReader.Create("_dc-vehicles.txt"))
```

Comment:
This file name is different from that used in the Serialize method, preventing the deserialization from working.